packages/aws-cdk-lib/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -135,7 +136,16 @@ export class AwsLogDriver extends LogDriver {
       ? `${this.props.maxBufferSize.toBytes({ rounding: SizeRoundingBehavior.FLOOR })}b`
       : undefined;
 
-    this.logGroup.grantWrite(containerDefinition.taskDefinition.obtainExecutionRole());
+    // These policies are required for the Execution role to use awslogs driver.
+    // In cases where `addToExecutionRolePolicy` is not implemented,
+    // for example, when used from aws-batch construct,
+    // use `obtainExecutionRole` instead of `addToExecutionRolePolicy` to grant policies to the Execution role.
```

Comment:
Would explicitly changing the call to `addToPrincipalPolicy` throw an error in mentioned case of `aws-batch`?

Reply:
Is it `addToExecutionRolePolicy` instead of `addToPrincipalPolicy`?
If so, the test will fail like this.

![スクリーンショット 2024-01-09 10 42 52](https://github.com/aws/aws-cdk/assets/58683719/fb5c55d9-ea43-43d5-bbae-cb0fcd4a700d)

Reply:
Yes sorry I misread the function name.